android/KMEA/app/src/main/assets/keyboard.html

```diff
@@ -64,39 +64,41 @@
     // Update the KMW banner height
     function setBannerHeight(h) {
       var kmw=com.keyman.singleton;
+      var osk = kmw.osk;
       if (h > 0) {
-        var osk = kmw.osk;
-        osk.banner.height = Math.ceil(h / window.devicePixelRatio);
+        // The banner itself may not be loaded yet.  This will preemptively help set
+        // its eventual display height.
+        com.keyman.osk.Banner.DEFAULT_HEIGHT = h;
+        if(osk.banner.activeType != 'blank') {
+          osk.banner.height = h; //Math.ceil(h / window.devicePixelRatio);
+        }
       }
-      // Refresh KMW OSK
+
+      // Refresh KMW's OSK
       kmw.correctOSKTextSize();
     }
 
-    function setOskHeight(h) {
-      if(h > 0) {
-        oskHeight = h;
-      }
-      var kmw=window['keyman'];
+    function setOskHeight(height) {
+      var kmw=com.keyman.singleton;
+      oskHeight = height;
       if(kmw && kmw.core && kmw.core.activeKeyboard) {
         kmw.core.activeKeyboard.refreshLayouts();
       }
-      kmw['correctOSKTextSize']();
+      kmw.osk.show(true);
+      kmw.correctOSKTextSize();
     }
 
     function setOskWidth(w) {
-      if(w > 0) {
-        oskWidth = w;
-      }
+      oskWidth = w;
```

Comment:
Also from iOS
https://github.com/keymanapp/keyman/blob/8843828879fedb3a9f6302aa115a8b08e7994982/ios/engine/KMEI/KeymanEngine/resources/Keyman.bundle/Contents/Resources/keyboard.html#L99-L107

Reply:
Although that snippet from iOS is for setting height, whereas this is width...?

Reply:
Oops.  here's setOskWidth

https://github.com/keymanapp/keyman/blob/f29c71d988e958b6dacb10f7c9be08b642864799/ios/engine/KMEI/KeymanEngine/resources/Keyman.bundle/Contents/Resources/keyboard.html#L109-L111